### PR TITLE
Fixes 1864 - Message forwarding room list filtering and pagination problems

### PR DESCRIPTION
**Pull request by stefanceriu**
* use a separate room summary provider instance just for message forwarding
* filter rooms through the rust sdk
* add pagination support

### CODE REVIEW

File: ElementX/Sources/Other/HTMLParsing/DTHTMLElement+AttributedStringBuilder.swift [hunk 1]
```diff
@@ -64,13 +64,11 @@ public extension DTHTMLElement {
...
-    private static var allowedHTMLTags = {
-        ["font", // custom to matrix for IRC-style font coloring
-         "del", // for markdown
-         "body", // added internally by DTCoreText
-         "mx-reply",
-         "h1", "h2", "h3", "h4", "h5", "h6", "blockquote", "p", "a", "ul", "ol",
-         "nl", "li", "b", "i", "u", "strong", "em", "strike", "code", "hr", "br", "div",
-         "table", "thead", "caption", "tbody", "tr", "th", "td", "pre"]
-    }()
+    private static var allowedHTMLTags = ["font", // custom to matrix for IRC-style font coloring
```
Comment (pixlwave) [inline ElementX/Sources/Other/HTMLParsing/DTHTMLElement+AttributedStringBuilder.swift:67]: Good spot. I think this can be a `let` too?
```suggestion
    private static let allowedHTMLTags = ["font", // custom to matrix for IRC-style font coloring
```

File: ElementX/Sources/Screens/MessageForwardingScreen/View/MessageForwardingScreen.swift [hunk 2]
```diff
@@ -22,10 +22,18 @@ struct MessageForwardingScreen: View {
     var body: some View {
         Form {
             Section {
-                ForEach(context.viewState.visibleRooms) { room in
+                ForEach(context.viewState.rooms) { room in
                     MessageForwardingRoomCell(room: room, context: context)
                         .buttonStyle(FormButtonStyle(accessory: .singleSelection(isSelected: context.viewState.selectedRoomID == room.id)))
                 }
+            } header: {
```
Comment (pixlwave) [inline ElementX/Sources/Screens/MessageForwardingScreen/View/MessageForwardingScreen.swift:29]: Is it worth adding a note for the future that whenever we drop iOS 16, these can be replaced by the new scrollview APIs?

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 3]
```diff
@@ -35,10 +35,11 @@ class ClientProxy: ClientProxyProtocol {
     private var syncService: SyncService?
     private var syncServiceStateUpdateTaskHandle: TaskHandle?
     
-    var roomSummaryProvider: RoomSummaryProviderProtocol?
-    var inviteSummaryProvider: RoomSummaryProviderProtocol?
+    private(set) var roomSummaryProvider: RoomSummaryProviderProtocol?
+    private(set) var messageForwardingRoomSummaryProvider: RoomSummaryProviderProtocol?
```
Comment (pixlwave) [inline ElementX/Sources/Services/Client/ClientProxy.swift:39]: Maybe a comment on this one that explains why we have both using the same list?

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift [hunk 4]
```diff
@@ -22,10 +22,11 @@ class RoomSummaryProvider: RoomSummaryProviderProtocol {
     private let roomListService: RoomListServiceProtocol
     private let eventStringBuilder: RoomEventStringBuilder
     private let name: String
+    private let shouldUpdateVisibleRange: Bool
     private let notificationSettings: NotificationSettingsProxyProtocol
     private let backgroundTaskService: BackgroundTaskServiceProtocol
     
-    private let roomListPageSize = 200
+    private let roomListPageSize = 40
```
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:29]: Was this change intentional??

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift [hunk 5]
```diff
@@ -59,12 +60,14 @@ class RoomSummaryProvider: RoomSummaryProviderProtocol {
     init(roomListService: RoomListServiceProtocol,
          eventStringBuilder: RoomEventStringBuilder,
          name: String,
+         shouldUpdateVisibleRange: Bool = false,
```
Comment (pixlwave) [inline ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:63]: Two thoughts:
- Might also be useful to document why you would pass this in as `true`/`false`?
- Shouldn't the default value be `true` here?

File: ElementX/Sources/Screens/MessageForwardingScreen/MessageForwardingScreenViewModel.swift [hunk 6]
```diff
@@ -91,4 +104,22 @@ class MessageForwardingScreenViewModel: MessageForwardingScreenViewModelType, Me
         
         MXLog.verbose("Finished updating rooms")
     }
+    
+    /// The actual range values don't matter as long as they contain the lower
+    /// or upper bounds
```
Comment (pixlwave) [inline ElementX/Sources/Screens/MessageForwardingScreen/MessageForwardingScreenViewModel.swift:109]: Should we add some more detail to this comment? Coming to it fresh, it doesn't particularly explain why this is the case.

**Review by pixlwave — APPROVED** [review]
LGTM

File: ElementX/Sources/Other/HTMLParsing/DTHTMLElement+AttributedStringBuilder.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Other/HTMLParsing/DTHTMLElement+AttributedStringBuilder.swift:67]: Thanks, all SwiftFormat's doing 😅

File: ElementX/Sources/Screens/MessageForwardingScreen/View/MessageForwardingScreen.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/MessageForwardingScreen/View/MessageForwardingScreen.swift:29]: Yep, good idea, let's keep it consistent

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift (hunk 4 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:29]: Definitely not 🙈

File: ElementX/Sources/Services/Client/ClientProxy.swift (hunk 3 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/Client/ClientProxy.swift:39]: Do you think that's really necessary, they different instances of the same list which I assumed makes perfect sense without any extra comments 🤔

File: ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift (hunk 5 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/Room/RoomSummary/RoomSummaryProvider.swift:63]: > Might also be useful to document why you would pass this in as true/false?

Will do

> Shouldn't the default value be true here?

The only summary provider that should actually worry about this is the room list one so I figured I wouldn't bother all the rest

File: ElementX/Sources/Services/Client/ClientProxy.swift (hunk 3 above)
Comment (pixlwave) [reply ElementX/Sources/Services/Client/ClientProxy.swift:39]: If you hadn't seen this PR it would be a valid question to ask *why* though? Just a short comment like so would be enough
```
/// A separate provider for message forwarding to avoid showing a filtered room list.
```

Comment (stefanceriu) [reply ElementX/Sources/Services/Client/ClientProxy.swift:39]: Fair
